translate/automl/src/main/java/com/google/cloud/translate/automl/CreateDataset.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2019 LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.translate.automl;
+
+// [START automl_translate_create_dataset]
+import com.google.api.gax.longrunning.OperationFuture;
+import com.google.cloud.automl.v1.AutoMlClient;
+import com.google.cloud.automl.v1.Dataset;
+import com.google.cloud.automl.v1.LocationName;
+import com.google.cloud.automl.v1.OperationMetadata;
+import com.google.cloud.automl.v1.TranslationDatasetMetadata;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+
+class CreateDataset {
+
+  // Create a dataset
+  static void createDataset(String projectId, String displayName) {
+    // String projectId = "YOUR_PROJECT_ID";
+    // String displayName = "YOUR_DATASET_NAME";
+
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the "close" method on the client to safely clean up any remaining background resources.
+    try (AutoMlClient client = AutoMlClient.create()) {
+      // A resource that represents Google Cloud Platform location.
+      LocationName projectLocation = LocationName.of(projectId, "us-central1");
+
+      // Specify the source and target language.
+      TranslationDatasetMetadata translationDatasetMetadata =
+          TranslationDatasetMetadata.newBuilder()
+              .setSourceLanguageCode("en")
+              .setTargetLanguageCode("ja")
+              .build();
+      Dataset dataset =
+          Dataset.newBuilder()
+              .setDisplayName(displayName)
+              .setTranslationDatasetMetadata(translationDatasetMetadata)
+              .build();
+      OperationFuture<Dataset, OperationMetadata> future =
+          client.createDatasetAsync(projectLocation, dataset);
+
+      Dataset createdDataset = future.get();
+
+      // Display the dataset information.
+      System.out.format("Dataset name: %s\n", createdDataset.getName());
+      System.out.format(
+          "Dataset id: %s\n",
+          createdDataset.getName().split("/")[createdDataset.getName().split("/").length - 1]);
```

Comment:
ok, but could you clean this up a bit and teach something here?

Reply:
Would something like this be good?
```
      // To get the dataset id, you have to parse it out of the `name` field. As dataset Ids are
      // required for other methods.
      // Name Form: `projects/{project_id}/locations/{location_id}/datasets/{dataset_id}`
      int lastIndex = createdDataset.getName().split("/").length - 1;
      String datasetId = createdDataset.getName().split("/")[lastIndex];
      System.out.format("Dataset id: %s\n", datasetId);
```

Reply:
I was thinking more like:
```
String[] names = createdDataset.getName().split("/");
String datasetId = names[names.length - 1];
```

Reply:
👍 I will probably keep the comment in there too.